content/docs/start/experiments/building-pipelines.md

```diff
@@ -0,0 +1,184 @@
+---
+title: 'Get Started: Building Pipelines'
+description:
+  'Split your workflow into stages and build a pipeline by connecting
+  dependencies and outputs.'
+---
+
+# Get Started: Building Pipelines
+
+Eventually, managing your notebook cells may start to feel fragile, and you may
+want to structure your project and code. When you are ready to
+[migrate from notebooks to scripts](https://towardsdatascience.com/from-jupyter-notebook-to-sc-582978d3c0c),
+DVC <abbr>Pipelines</abbr> can help you standardize your workflow following
+software engineering best practices:
+
+- **Modularization**: split the different logical steps in your notebook into
+  separate scripts.
+
+- **Parametrization**: adapt your scripts to decouple the configuration from the
+  source code.
+
+## Creating stages
+
+In our example repo, we first extract data preparation from the
+[original notebook](https://github.com/iterative/example-get-started-experiments/blob/main/notebooks/TrainSegModel.ipynb)
+into
+[`data_split.py`](https://github.com/iterative/example-get-started-experiments/blob/main/src/data_split.py).
+We parametrize this script by reading parameters from
+[`params.yaml`](https://github.com/iterative/example-get-started-experiments/blob/main/params.yaml):
+
+```python
+from ruamel.yaml import YAML
+
+yaml = YAML(typ="safe")
+
+def data_split():
+    params = yaml.load(open("params.yaml", encoding="utf-8"))
+...
+```
+
+You can use `dvc stage add` to transform a script into a <abbr>stage</abbr>:
+
+```cli
+$ dvc stage add --name data_split \
+  --params base,data_split \
+  --deps data/pool_data --deps src/data_split.py \
+  --outs data/train_data --outs data/test_data \
+  python src/data_split.py
+```
+
+A `dvc.yaml` file is generated. It includes information about the command you
+want to run (`python src/data_split.py`), its <abbr>dependencies</abbr>,
+<abbr>parameters</abbr>, and <abbr>outputs</abbr>:
+
+```yaml
+stages:
+  prepare:
+    cmd: python src/data_split.py
+    deps:
+      - src/data_split.py
+      - data/pool_data
+    params:
+      - base
+      - data_split
+    outs:
+      - data/train_data
+      - data/test_data
+```
+
+`dvc exp run` will run all stages in the `dvc.yaml` file.
```

Comment:
> Not blocking: 
> 
> Do you think is worth adding a ```console block showing a simplified output (like the `Running stage `data_split`...)
> 
> Commenting mostly because of the paragraph bellow where you introduce the skip/cache

_Originally posted by @daavoo in https://github.com/iterative/dvc.org/pull/4425#discussion_r1147312631_

Reply:
Pushed some changes related to your feedback. PTAL. Not sure if it is getting too long/detailed already.

Reply:
> Not sure if it is getting too long/detailed already.

The changes look good to me. And it is getting too long indeed but I don't see what we can omit without losing too much context.

This is why I originally couldn't stop myself from decoupling `Pipelines` completely 😅 .

Maybe we can revisit the idea as a follow-up.